src/devices/Ht1632/Ht1632.cs

```diff
@@ -0,0 +1,189 @@
+    /// HT1632C
+    /// 32x8 and 24x16 LED Driver
+    /// </summary>
+    public class Ht1632 : IDisposable
+    {
+        // Datasheet: https://www.holtek.com/documents/10179/116711/HT1632D_32D-2v100.pdf
+
+        /// <summary>
+        /// Set if system oscillator is on. (default is off)
+        /// </summary>
+        public bool Enabled { set => WriteCommand(value ? Command.SYS_EN : Command.SYS_DIS); }
+
+        /// <summary>
+        /// Set if LED duty cycle generator is on (default is off)
+        /// </summary>
+        public bool LedOn { set => WriteCommand(value ? Command.LED_On : Command.LED_Off); }
+
+        /// <summary>
+        /// Set if blinking function is on (default is off)
+        /// </summary>
+        public bool Blink { set => WriteCommand(value ? Command.BLINK_On : Command.BLINK_Off); }
+
+        /// <summary>
+        /// Set clock mode (default is RC-Master)
+        /// </summary>
+        public ClockMode ClockMode { set => WriteCommand((Command)value); }
+
+        /// <summary>
+        /// Set COM option (default is N-MOS open drain output and 8 COM)
+        /// </summary>
+        public ComOption ComOption { set => WriteCommand((Command)value); }
+
+        /// <summary>
+        /// Set row PWM duty (1/16 to 16/16, default is 16)
+        /// </summary>
+        public byte PwmDuty { set => WriteCommand(s_pwmCommandMap[value]); }
+
+        private static readonly Dictionary<byte, Command> s_pwmCommandMap = new Dictionary<byte, Command>()
+        {
+            { 1, Command.PWM_Duty_1 },
+            { 2, Command.PWM_Duty_2 },
+            { 3, Command.PWM_Duty_3 },
+            { 4, Command.PWM_Duty_4 },
+            { 5, Command.PWM_Duty_5 },
+            { 6, Command.PWM_Duty_6 },
+            { 7, Command.PWM_Duty_7 },
+            { 8, Command.PWM_Duty_8 },
+            { 9, Command.PWM_Duty_9 },
+            { 10, Command.PWM_Duty_10 },
+            { 11, Command.PWM_Duty_11 },
+            { 12, Command.PWM_Duty_12 },
+            { 13, Command.PWM_Duty_13 },
+            { 14, Command.PWM_Duty_14 },
+            { 15, Command.PWM_Duty_15 },
+            { 16, Command.PWM_Duty_16 },
+        };
+
+        private readonly Ht1632PinMapping _pinMapping;
+        private readonly int _cs;
+        private readonly int _wr;
+        private readonly int _data;
+        private readonly bool _shouldDispose;
+
+        private GpioController? _controller;
+
+        /// <summary>
+        /// HT1632C 32x8 and 24x16 LED Driver
+        /// </summary>
+        /// <param name="pinMapping">The pin mapping to use by the binding.</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling.</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance.</param>
+        public Ht1632(Ht1632PinMapping pinMapping, GpioController? gpioController = null, bool shouldDispose = true)
+        {
```

Comment:
This seems very much like SPI. I guess for performance it will be better. You can use Software SPI as well as this implementation seems very much like this. I may have missed something but what's the key difference here and why would SPI not be appropriate? I see some elements that may be sent less than by a group of 8 but what the doc says about that if we more clocks are sent?

Reply:
I've considered SPI, but it doesn't require a complete 8-bit set of bits. 
The command format is like "XXX-XXXXXXXX-X"
The data format is like "XXX-XXXXXX-XXXX-XXXX-XXXX-..."
not 8-bit bytes
Using today's SPI interface can be difficult, or performance issues

Reply:
ok, fair enough, let's leave it with GPIO for now.